src/odemis/acq/align/test/roi_autofocus_test.py

```diff
@@ -0,0 +1,137 @@
+        cls.ccd = model.getComponent(role="ccd")
+        cls.focus = model.getComponent(role="focus")
+        cls.stage = model.getComponent(role="stage")
+
+        # Assumes the stage is referenced
+        cls.init_pos = (cls.stage.position.value["x"], cls.stage.position.value["y"])
+        cls.width, cls.height = cls.ccd.shape[:2]
+        cls.px_size = cls.ccd.getMetadata().get(model.MD_PIXEL_SIZE)
+
+        # Assume initial focus position is good
+        # The initial position comes from manual focussing the centre of overview image
+        current_focus = cls.focus.position.value["z"]  # assume manual focus before running the script
+        # Allow a range of +/- 30 µm around the focused position
+        cls.focus_range = (current_focus - 30.0e-6, current_focus + 30.0e-6)
+
+    @classmethod
+    def tearDownClass(cls):
+        logging.debug("moving back to initial position")
+        cls.stage.moveAbs({"x": cls.init_pos[0], "y": cls.init_pos[1]}).result()
+
+    def test_autofocus_in_roi(self):
+        """
+        Tests the autofus in roi is running and returns the focus points:
+        """
+        init_pos = self.init_pos
+        px_size = self.px_size
+        width = self.width
+        height = self.height
+        n_tiles = (3, 3)
+        overlap = 0.2
+
+        xmin = init_pos[0] - (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymin = init_pos[1] - (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+        xmax = init_pos[0] + (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymax = init_pos[1] + (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+
+        bbox = (xmin, ymin, xmax, ymax)
+
+        f = autofocus_in_roi(bbox, self.stage, self.ccd, self.focus, self.focus_range, n_tiles[0], n_tiles[1], 0)
+
+        # Test if the autofocus in roi is running
+        time.sleep(2)
+        self.assertTrue(f.running())
+
+        # The returned focus points should be minimum three
+        focus_points = f.result()
+        self.assertGreaterEqual(len(focus_points), 3)
+
+        # Test if the autofocus task is done
+        self.assertTrue(f.done())
+
+    def test_cancel_autofocus_in_roi(self):
+        """
+        Tests cancelling of autofocus
+        """
+        init_pos = self.init_pos
+        px_size = self.px_size
+        width = self.width
+        height = self.height
+        n_tiles = (3, 3)
+        overlap = 0.2
+
+        xmin = init_pos[0] - (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymin = init_pos[1] - (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+        xmax = init_pos[0] + (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymax = init_pos[1] + (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+
+        bbox = (xmin, ymin, xmax, ymax)
+
+        f = autofocus_in_roi(bbox, self.stage, self.ccd, self.focus, self.focus_range, n_tiles[0], n_tiles[1],
+                             0)  # TODO add proper confidence value
```

Comment:
Since you are only testing that it is running I think it is fine to use a confidence_level of 0. It just means that none of the focus points will be discarded